nselib/url.lua

```diff
@@ -177,8 +178,22 @@ function parse(url, default)
     parsed.params = p
     return ""
   end)
+
   -- path is whatever was left
   parsed.path = url
+
+  -- Checks for folder route and extension
+  if(string.match(parsed.path, "/$")) then
```

Comment:
Using `parsed.path:sub(-1) == "/"` is less expensive than `string.match(parsed.path, "/$")`.

Even if you do wish to use pattern matching, consider using `string.find` instead of `string.match` if you do not need the matched string. This way the intent is slightly more clear.